openshift/system/entrypoint.sh

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
```

Comment:
Have you read this? https://stackoverflow.com/questions/19622198/what-does-set-e-mean-in-a-bash-script
And this? http://mywiki.wooledge.org/BashFAQ/105
It might not be the best solution

Reply:
I always use `-e` and I started to do so before bash mode or whatever. It is from experience (while I was writing long things in bash) that error detection code is much less verbose in this way. Some false positives that are usually easy to catch in the beginning. You can see the results by yourself - how much time we lost for a few commands that failed while nobody was watching.

Also much less error prone than trying to spot all places things may fail.

What is your suggested alternative?

P.S. although I didn't see any pipes, pipefail is also good to have usually.

Reply:
> What is your suggested alternative?

I don't have any. The stackoverflow post mentions `trap 'do_something' ERR` But I haven't tried it.

Reply:
> http://mywiki.wooledge.org/BashFAQ/105
This one is particularly interesting, nice read.

What I take from it: you can use `set -e` if you know what you are doing, do not rely on it 100%, and are aware of potential false positives.

But actually, as there are a few if-else statements, I think that indeed we need to be careful with it. I'll take a closer look.

Reply:
Within the `if` condition, it doesn't exit the program. But sure, that's good to check anything that looks fishy to you.